### DISCUSSION TITLE
Using the extra parameter in the FastApi constructor

### DISCUSSION THREAD

**Discussion by ddvir** [fastapi/fastapi#5802] [Questions] [answered]
The FastApi constructor takes a parameter called extra. I would like to use it to add additional Open API specification information like custom extensions. Unfortunately, it appears this parameter is not used anywhere in the code base.

All the HTTP verb decorators accept an openapi_extra parameter that can be used exactly as I described. When passing a dictionary it gets serialized into the Open API specification returned by running curl against: http://localhost:8000/openapi.json .

I've achieved the desired result right now by overriding the openapi method that is part of the FastApi class but this feels cumbersome and unnecessary. Is it possible to add the serialization of the self.extra member into the get_openapi method?

**Comment by caeser1996** [comment]
Yes, it is possible to add the serialization of the self.extra member into the get_openapi method of the FastAPI class.

You could subclass the FastAPI class and override the get_openapi method to include the serialization of the self.extra member. Here's an example:
```
from fastapi import FastAPI

class CustomFastAPI(FastAPI):
    def __init__(self, *args, **kwargs):
        self.extra = kwargs.pop("extra", None)
        super().__init__(*args, **kwargs)

    async def get_openapi(self, **kwargs):
        openapi = await super().get_openapi(**kwargs)
        if self.extra:
            openapi["x-extra"] = self.extra
        return openapi

```
In this example, the CustomFastAPI class takes a extra parameter in its constructor, just like the FastAPI class. The get_openapi method is then overridden to include the serialization of the self.extra member in the returned OpenAPI specification.

You can then use the CustomFastAPI class in place of the FastAPI class to take advantage of this added functionality.

**Comment by YuriiMotov** [comment, marked as answer]
From docs:
> Extra keyword arguments to be stored in the app, not used by FastAPI anywhere.

So, it's not related to `openapi_extra`.
So, adding this to openapi schema manually is the only way
